pkg/sanitization/aws/s3.go

```diff
@@ -6,19 +6,54 @@ import (
 	"github.com/klothoplatform/klotho/pkg/sanitization"
 )
 
-// EnvVarKeySanitizer returns a sanitized environment key when applied.
 var S3BucketSanitizer = sanitization.NewSanitizer(
 	[]sanitization.Rule{
-		// replace "-" or whitespace with "_"
+		// must not contain non-alphanumeric characters other than "-" or "."
+		// uppercase characters will be converted to lowercase
 		{
-			Pattern:     regexp.MustCompile(`[^a-z0-9.-]`),
+			Pattern:     regexp.MustCompile(`[^a-zA-Z\d.-]`),
 			Replacement: "-",
+			Lowercase:   true,
+		},
+		{
+			// must not start with "xn--"
+			Pattern:     regexp.MustCompile(`^(xn--)+`),
+			Replacement: "",
+		},
+		{
+			// must start and end with a letter or number
+			Pattern:     regexp.MustCompile(`^[^a-z\d]+|[^a-z\d]+$`),
+			Replacement: "",
+		},
+		{
+			// must not contain repeated periods
+			Pattern:     regexp.MustCompile(`\.\.+`),
+			Replacement: ".",
+		},
+		{
+			// must not be formatted as an IP address (for example, 192.168.5.4)
```

Comment:
Just to double check, do we need to also check IPV6?

(If so, I'd recommend not doing that via regex, and just trying to parse it using some real library)